docs/src/architecture/08_concepts/signed_doc/cddl/signed_document.cddl

```diff
@@ -0,0 +1,144 @@
+; Catalyst Signed Document.
+;
+; A specific implementation of a COSE-SIGN data objects
+; used by the Catalyst project to encapsulate and authenticate
+; documents used within the system.
+;
+; See: https://datatracker.ietf.org/doc/html/rfc9052#name-signing-with-one-or-more-si
+
+
+; Catalyst Signed Document data object.
+signed_document = COSE_Sign
+
+; COSE-SIGN data object
+COSE_Sign = [
+  COSE_Document_Headers,
+  payload : bstr / nil,
+  signatures : [+ COSE_Signature]
+]
+
+; COSE Document headers (only protected headers are used)
+COSE_Document_Headers = (
+  protected   : bstr .cbor COSE_Document_Header_Map,
+  unprotected : { COSE_Generic_Headers } ; Unused and ignored
+)
+
+; COSE Document Header Map
+COSE_Document_Header_Map = {
+  COSE_Document_Standard_Headers,
+  Signed_Document_Metadata_Headers,
+  COSE_Generic_Headers
+}
+
+; COSE Standard headers used by a Document
+COSE_Document_Standard_Headers = (
+  3 => iana_media_types
+  ?"content-encoding" => http_content_encoding
+)
+
+; Supported Content Media Types
+iana_media_types = (
+  "application/json" /
+  "application/schema+json" /
```

Comment:
This is not yet iana registered media type, maybe rename this into just `media_types`.
Another thing is that such types like `application/json`, `application/cbor` etc. which are COAP format registered, they could be encoded by the tag value specified by this registry

https://www.iana.org/assignments/core-parameters/core-parameters.xhtml
https://datatracker.ietf.org/doc/html/rfc8152#page-84

Reply:
I can rename it.  No problem.
I would rather not support CoAp because it just adds a complexity with no functional value.  Ie, I can use an integer OR a string to define the same thing.  Much easier to just only recognize the string.  Especially as we have content types which are not standard.  
We aren't meaningfully saving space by saving 16 bytes or something in a document that's probably a few kb or bigger.

Reply:
>I can use an integer OR a string to define the same thing
Yes correct, but the problem is that the coset library is using CoAp by default which is actually used in our current implementation. 
>Much easier to just only recognize the string
But if we will recognise only string it means we are not fully implementing COSE decoding logic, could it be the problem for us ?